### PR TITLE
4.x: Introduction of HSON library to write and parse Helidon metadata files.

### PR DESCRIPTION
**Pull request by tomas-langer**
Prerequisite for service registry and config reference documentation.

### CODE REVIEW

File: metadata/hson/src/test/java/io/helidon/metadata/hson/HsonTest.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import static io.helidon.common.testing.junit5.OptionalMatcher.optionalPresent;
+import static io.helidon.common.testing.junit5.OptionalMatcher.optionalValue;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class HsonTest {
```
Comment (romain-grecourt) [inline metadata/hson/src/test/java/io/helidon/metadata/hson/HsonTest.java:39]: should be package private.

File: metadata/hson/src/test/java/io/helidon/metadata/hson/TestExistingTypes.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import static io.helidon.common.testing.junit5.OptionalMatcher.optionalEmpty;
+import static io.helidon.common.testing.junit5.OptionalMatcher.optionalPresent;
+import static io.helidon.common.testing.junit5.OptionalMatcher.optionalValue;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+
+public class TestExistingTypes {
```
Comment (romain-grecourt) [inline metadata/hson/src/test/java/io/helidon/metadata/hson/TestExistingTypes.java:35]: Should be package private

File: metadata/hson/src/test/java/io/helidon/metadata/hson/HsonTest.java (hunk 1 above)
Comment (tomas-langer) [reply metadata/hson/src/test/java/io/helidon/metadata/hson/HsonTest.java]: Fixed

File: metadata/hson/src/test/java/io/helidon/metadata/hson/TestExistingTypes.java (hunk 2 above)
Comment (tomas-langer) [reply metadata/hson/src/test/java/io/helidon/metadata/hson/TestExistingTypes.java]: Fixed (and renamed)

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**